projects/freertos/cmd_channel_freertos.c

```diff
@@ -26,7 +25,7 @@ int cmd_channel_freertos_receive_packet (QueueHandle_t rx_queue, struct cmd_pack
 		return CMD_CHANNEL_INVALID_ARGUMENT;
 	}
 
-	status = xQueueReceive (rx_queue, packet, timeout);
+	status = xQueueReceive (I2CRequestQueue, packet, timeout);
```

Comment:
Global variables are typically avoided to allow for flexibility in the implementation.  Having a global variable here prevents this code from being reused in scenarios where you might have multiple command channels.

Reply:
The current implementation of changing the first argument to
    cmd_channel_freertos_receive_packet()/cmd_channel_freertos_send_packet()
    from a struct *cmd_channel (passed from
    cmd_channel_receive_and_process()) to a QueueHandle_t results in seg faults.
    
    Instead let's create a global queue that can be accessed from anywhere
    to allow data to be stored.
    
    The queue being global allows it to be accessed from interrupt service
    routines (ISRs) which is generally where the data will be processed, at
    least for receives.

Reply:
What is currently there doesn't work. This is just getting it to work. I agree passing in a queue is better, but that would require changes to the `mctp_cmd_task_loop()` function and a way to handle that in an interrupt handler. I think that should be left for a future patch.

Reply:
It is certainly incorrect to assert what is here doesn't work, because this is actively being using in the current Microsoft implementation.  What I think is missing is a clear example of how these pieces are interrelated, since we don't have a reference port available at this time.

The first thing to point out is that cmd_channel_freertos does not represent an implementation of the cmd_channel API. 
 Perhaps the module could have been named better, but there is no structure of this type defined so it cannot implement the API.  These functions are meant to be helper functions that an implementing cmd_channel could use in a FreeRTOS environment.  However, these functions cannot be directly used as an implementation of the cmd_channel APIs because they do not match the API signature.  There will need to be a wrapper in the cmd_channel implementation that provides the current queue to these calls.

A cmd_channel implementation is expected to be some kind of bus driver (e.g. I2C) that will send and receive packets.  That instance would own Rx and/or Tx FreeRTOS queues that it would pass into these functions as part of receive_packet/send_packet API processing, as appropriate.  Any ISRs would be part of this driver implementation and would have access to the instance structure with the necessary queue(s), so no global piece is required.

Reply:
Ah, I see.

That is very confusing because there is a `cmd_channel_init()` in `cmd_channel.c` but then `cmd_channel_freertos.c` doesn't match with it.

Would you be open to me adding an example `cmd_channel_receive_packet()` function that matches the `cmd_channel` API and calls `cmd_channel_freertos_receive_packet()`?

Reply:
Sure.  That sounds like a good idea.

Reply:
Done! Let me know what you think.